src/nix/version.rs

```diff
@@ -0,0 +1,78 @@
+//! Rust module for `nix --version`
+use leptos::*;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+#[cfg(feature = "ssr")]
+use tracing::instrument;
+
+/// Nix version as parsed from `nix --version`
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub struct NixVersion {
+    major: u32,
+    minor: u32,
+    patch: u32,
+}
+
+/// Parse the string output of `nix --version` into a [NixVersion]
+pub fn parse_nix_version(version_str: &str) -> Option<NixVersion> {
+    let components: Vec<&str> = version_str.split_whitespace().collect();
+
+    if components.len() >= 2 {
+        let version = components[components.len() - 1];
+        let version_parts: Vec<&str> = version.split('.').collect();
+
+        if version_parts.len() == 3 {
+            let major = version_parts[0].parse().ok()?;
+            let minor = version_parts[1].parse().ok()?;
+            let patch = version_parts[2].parse().ok()?;
+
+            Some(NixVersion {
+                major,
+                minor,
+                patch,
+            })
+        } else {
+            None
+        }
+    } else {
+        None
+    }
+}
+
+/// Get the output of `nix --version`
+#[cfg(feature = "ssr")]
+#[instrument(name = "version")]
+pub async fn run_nix_version() -> Result<NixVersion, ServerFnError> {
+    use tokio::process::Command;
+    let mut cmd = Command::new("nix");
+    cmd.arg("--version");
+    let stdout: Vec<u8> = crate::command::run_command(&mut cmd).await?;
+    let v = parse_nix_version(std::str::from_utf8(&stdout).unwrap()).unwrap();
```

Comment:
We should not use `unwrap`, [it panics](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap). Also, use `from_utf8_lossy` - since we don't care about utf8 errors in version strings.